### PR TITLE
resources: add more links to record serialization

### LINKED ISSUE #438: resource: serialisation should include links
The REST API serialization should include links to:

- The access links endpoint (if user has manage access)
- The latest version api (all) - GET /api/records/:id/latest
- The latest version html (all) - GET /records/:id/latest
- The search all version (all) - GET /records/:id/versions

```python
{
    "id": "...",
    "links": {
        # ...
        "access_links": "https://localhost:5000/api/records/:id/access/links",
        "versions": "https://localhost:5000/api/records/:id/versions",
        "latest": "https://localhost:5000/api/records/:id/versions/latest",
        "latest_html": "https://localhost:5000/records/:id/latest",
    }
}
```

### PR DESCRIPTION
**Pull request by max-moser**
closes #438 
* add links to the secret links endpoint, as well as versions-related endpoints

### CODE REVIEW

File: invenio_rdm_records/resources/config.py [hunk 1]
```diff
@@ -28,17 +28,29 @@
...
 })
 
 
 DraftLinks = LinksSchema.create(links={
     "self": ItemLink(template="/api/records/{pid_value}/draft"),
     "self_html": ItemLink(template="/uploads/{pid_value}"),
     "files": ItemLink(template="/api/records/{pid_value}/draft/files"),
+    "versions": ItemLink(template='/api/records/{pid_value}/versions'),
+    "latest": ItemLink(template='/api/records/{pid_value}/versions/latest'),
+    "latest_html": ItemLink(template='/records/{pid_value}/versions/latest'),
```
Comment (fenekku) [inline invenio_rdm_records/resources/config.py:46]: These might point to something not existing right? Is that ok?

File: invenio_rdm_records/resources/config.py [hunk 2]
```diff
@@ -28,17 +28,29 @@
...
+    "access_links": ItemLink(
+        template='/api/records/{pid_value}/access/links', permission="manage"
```
Comment (fenekku) [inline invenio_rdm_records/resources/config.py:48]: might be nice to test for this.

**Review by fenekku — APPROVED** [review]

File: invenio_rdm_records/resources/config.py (hunk 1 above)
Comment (max-moser) [reply invenio_rdm_records/resources/config.py:46]: yes indeed
right now, those links would simply 404, which isn't *entirely* wrong, but it'd probably be saner to only include them if it wouldn't lead to a 404 (more effort to make this conditional, of course), or simply remove them

Comment (fenekku) [reply invenio_rdm_records/resources/config.py:46]: We have this: https://github.com/inveniosoftware/marshmallow-utils/blob/master/marshmallow_utils/fields/links.py#L52 for conditional rendering.

But it's probably fine for now...

**Review by fenekku — APPROVED** [review]

**Pull request merged**